hal/armv7m/imxrt/bootloader.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * ROM-API bootloader interface i.MX RT series
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _BOOTLOADER_H_
+#define _BOOTLOADER_H_
+
+
+/* Serial downloader media */
+#define BOOT_DOWNLOADER_AUTO (0x10uL << 16)
+#define BOOT_DOWNLOADER_USB  (0x11uL << 16)
+#define BOOT_DOWNLOADER_UART (0x12uL << 16)
+
+#define BOOT_IMAGE_SELECT(n) ((n)&3uL)
```

Comment:
clang-format is confused sometimes
```suggestion
#define BOOT_IMAGE_SELECT(n) ((n) & 3uL)
```

Reply:
Done.